rfcs/0000-standardize_the_kuadrant_spec.md

```diff
@@ -0,0 +1,238 @@
+# Standardize the Kuadrant Spec
+
+- Feature Name: Standardize the Kuadrant Spec
+- Start Date: 10/01/2025
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC is trying to describe a common API design that can be used when adding feature to Kuadrant that are configurable thought the Kuadrant CR.
+
+# Motivation
+[motivation]: #motivation
+
+As we are move forward with the development of Kuadrant, we are wanting to add new features to the product.
+Some of these features will be configurable by the user, the Kuadrant CR is an expected place for this to happen. 
+Before starting to add these configurable features, a standardize way of defining features in the Kuadrant CR should be agreed upon. 
+This RFC lays the groundwork for an API design which the new features will follow.
+By having a standardized API design, future development can move faster as the API design will have the groundwork already done.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The core of the design is made up of four elements. 
+`spec.feature_a` is the name of the feature that is being configured in that section.
+`spec.feature_a.configure` is a boolean field which tells the kuadrant operator to use the configuration defined in the Kuadrant CR for the feature. 
+This is done to allow us the flexibility to have features active by default.
+`spec.feature_a.enable` is a list of strings that represent the components the configuration will be applied too.
+Having it a list of strings allows us to extend or reduce the number of accepted values without needing to modify the CRD. 
+With the use of strings as values means keywords can also be defined, for example `"default"` could be shorthand for a sub set of components affected by the feature.
+`spec.feature_a.disable` is a list of strings that repenting the components that the configuration will *not* be applied too.
+Again shorthand keywords can be used here.
+```yaml
+spec:
+  feature_a:
+    configure: true # optional, boolean, defaults to false
+    enable: [] # optional, list of strings
+    disable: [] # optional, list of strings
```

Comment:
I think having both an 'enable' and 'disable' list would be confusing for the user.
Are they meant to be mutually exclusive?
Should only 1 or the other be set?
Or should I make sure all components are included across the union of both lists.

Reply:
When you say mutually exclusive, which do you mean? 
1. you can only have the `enable` or `disable` lists.
2. a value can only be in `enable` or `disable`, but not in both.

Reply:
Either.
Hence the confusion.
Also, what happens if i'm not exhaustive in what i put in both arrays?
For example:
```yaml
enable: ['authorino']
disable: ['limitador']
```
Do all **other** components have the feature enabled or disabled?

I'm not looking for an answer here (at least not yet).
Moreso highlighting that it's confusing to me.

Reply:
I will do a bit of rewording around this. I am guess that the `# optional` is also causing some of this confusion.